CONTRIBUTING.md

```diff
@@ -70,20 +71,73 @@ When you are contributing to changelog, please follow these suggestions:
 
 3. Once you have completed your work, create a Pull Request, ensuring that it meets the requirements listed below.
 
-### Testing
+### Requirements for Pull Requests
+
+* Please create Pull Requests against the `master` branch.
+* Please make sure that your code complies with [PEP8](https://www.python.org/dev/peps/pep-0008/).
+* One line should not contain more than 100 characters.
+* Make sure that new code is covered by a test case (new or existing one).
+* We don't like [spaghetti code](https://en.wikipedia.org/wiki/Spaghetti_code).
+* All the tests in the test-suite have to pass.
+
+## Development Environment
+
+### Requirements
+
+Currently, the development environment can be set up on __Linux__, and __MacOS__.
+Before beginning the setup, please ensure you have installed and set up the following on your machine:
+
+- Docker ([installation guide](https://docs.docker.com/install/))
+- make utility
+
+### Building a test image
+
+Open up a terminal and go to the cloned `release-bot` git repository.
+Execute the command
+
+```
+make image-test
+```
+
+It builds a Docker image using your local repository, with release-bot installed in it. This may take some time when building for the first time.
+
+To check if the build was successful, run  `docker images`. If the build was successful, an image named `release-bot-tests` will be present in the output list.
+
+### Testing new changes
+
+To test any new changes you made in the code, first build a test image using the above instructions, and then execute the following command
```

Comment:
There are also ways how to run it locally without having to build and run a container:
- `cd release-bot; python3 -m release_bot.releasebot --debug --configuration=path/to/conf.yaml`
- `cd release-bot; pip3 install -e .; release-bot --debug --configuration=path/to/conf.yaml`

Reply:
Alright, so I'll add the instructions for testing without instructions. In the current doc, it is recommended to use containers, shall I do the same?

Reply:
I added the direct installation method. I also updated the docker command, so that the image won't have to be built again and again. The command is a bit too long though. Can we add a new make recipe for this?